Jorge Osto - Project setup

Jorlost19 opened the pull request.

Jorlost19 committed:
Project setup

Jorlost19 committed:
Implemented GET & POST

Raymondgrc commented on friends/src/App.js:
```diff
@@ -1,28 +1,50 @@
+  }
+
+  componentDidMount() {
+    axios.get('http://localhost:5000/friends').then(response => this.setState({friendsList: [...response.data]}))
```
I recommend adding a catch statement to handle errors that may occur with the request.

Raymondgrc commented on friends/src/App.js:
```diff
@@ -1,28 +1,50 @@
+        age: this.state.age,
+        email: this.state.email
+      }
+      axios.post('http://localhost:5000/friends', friend).then(axios.get('http://localhost:5000/friends')
```
As we discussed, this will work fin without the get request in here.

Raymondgrc reviewed (commented):
Great job so far, can't wait to see what you come up with.

Jorlost19 committed:
added reactstrap
refactored app to use different components from reactstrap
almost done with update method in new structure
still having an issue trying to update my state with the array in props

Jorlost19 committed:
Fixed everything

Jorlost19 committed:
working in styles

Jorlost19 committed:
Added Table to display friends data

Raymondgrc commented on friends/src/App.js:
```diff
@@ -0,0 +1,98 @@
+import './App.css';
+import axios from 'axios';
+import styled from 'styled-components';
+import { Button, FormGroup, Form, Input, Tooltip, Row, Col, Container } from 'reactstrap';
```
If you like this sort of styling, I'd recommend taking a look at Material Ui

Raymondgrc commented on friends/src/App.js:
```diff
@@ -0,0 +1,98 @@
+
+  toggleToolTip = () => {
+    this.setState({
+    tooltipOpen: !this.state.tooltipOpen
```
I like this. Nice and simple.

Raymondgrc commented on friends/src/App.js:
```diff
@@ -0,0 +1,98 @@
+      e.preventDefault();
+      if(this.state.name && this.state.age && this.state.email) {
+          const friend = {
+          id: this.state.friendsList.length + 1,
```
Is setting an id necessary? The server.js provides an id and returns the entire friends object back.

Raymondgrc commented on friends/src/App.js:
```diff
@@ -0,0 +1,98 @@
+                            <FormContainer>
+                                <Form>
+                                    <FormGroup>
+                                        <Input placeholder="Insert Friend's Name" onChange={this.changeHandler} name="name" value={this.state.name}></Input>
```
Input is a self closing tag, I would amend these to prevent any accidents.

Raymondgrc commented on friends/src/components/FriendsList.js:
```diff
@@ -0,0 +1,112 @@
+
+        render() {
+        return  <>
+                    <Table>
```
I like how you rendered your data. Great job on the effort.

Jorlost19 committed:
Finished all stretch goals
Finished Routes, styling and features

Jorlost19 committed:
Added prop-types to the project
cleaned some unused code that was meant for testing

Jorlost19 committed:
Fixed 1 prop check
